### PR TITLE
[26] [UI] As a user, I can see my Profile screen 

### LINKED ISSUE #26: [UI] As a user, I can see my Profile screen
## Why

- After tapping the profile tab, I can see my profile and settings

## Acceptance Criteria

- Show title `My Account` on the navigation bar
- Show `Profile` view:
  - Avatar
  - Name
  - Email
- Show `My Order` view:
  - My orders cell
  - My returns cell
- Show `My Addresses` view:
  - Shipping address
  - Billing address
  - Tax Invoice Address
  - My Wishlist
  - Logout

## Resources

Screen Shot 2021-05-05 at 09 50 09

- [Design](https://www.figma.com/proto/GXmbnWUEH9ZkrUluaPyedn/e-commerce?node-id=44%3A1274&scaling=min-zoom)

### PR DESCRIPTION
**Pull request by nmint8m**
https://github.com/nimblehq/nimble-ecommerce-ios/issues/26

## What happened 👀

- After tapping the profile tab, I can see my profile and settings
- Show title `My Account` on the navigation bar
- Show `Profile` view:
  - Avatar
  - Name
  - Email
- Show `My Order` view:
  - My orders cell
  - My returns cell
- Show `My Addresses` view:
  - Shipping address
  - Billing address
  - Tax Invoice Address
  - My Wishlist
  - Logout

### CODE REVIEW

File: ecommerce-ios/Models/SettingType.swift [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+        switch self {
+        case .myOrder: return "screen-profile/icon-my-order"
+        case .myReturn: return "screen-profile/icon-my-return"
+        case .shippingAddress: return "screen-profile/icon-shipping-address"
+        case .billingAddress: return "screen-profile/icon-billing-address"
+        case .taxInvoiceAddress: return "screen-profile/icon-tax-invoice-address"
+        case .myWishlist: return "screen-profile/icon-my-wishlist"
+        case .logOut: return "screen-profile/icon-log-out"
+        }
+    }
```
Comment (vnntsu) [inline ecommerce-ios/Models/SettingType.swift]:
```suggestion
    var imageString: String {
        let dir = "screen-profile"
        switch self {
        case .myOrder: return "\(dir)/icon-my-order"
        case .myReturn: return "\(dir)/icon-my-return"
        case .shippingAddress: return "\(dir)/icon-shipping-address"
        case .billingAddress: return "\(dir)/icon-billing-address"
        case .taxInvoiceAddress: return "\(dir)/icon-tax-invoice-address"
        case .myWishlist: return "\(dir)/icon-my-wishlist"
        case .logOut: return "\(dir)/icon-log-out"
        }
    }
```

File: ecommerce-ios/Models/SettingSectionType.swift [hunk 2]
```diff
@@ -0,0 +1,25 @@
+//
+//  SettingSectionType.swift
+//  ecommerce-ios
+//
+//  Created by Nguyen M. Tam on 20/05/2021.
+//
+
+enum SettingSectionType: Int, CaseIterable {
```
Comment (vnntsu) [inline ecommerce-ios/Models/SettingSectionType.swift]:
```suggestion
enum SettingsSectionType: Int, CaseIterable {
```

File: ecommerce-ios/Models/SettingType.swift [hunk 3]
```diff
@@ -0,0 +1,35 @@
+//
+//  SettingType.swift
+//  ecommerce-ios
+//
+//  Created by Nguyen M. Tam on 20/05/2021.
+//
+
+enum SettingType: Int, CaseIterable {
```
Comment (vnntsu) [inline ecommerce-ios/Models/SettingType.swift]:
```suggestion
enum SettingsType: Int, CaseIterable {
```

File: ecommerce-ios/Sources/Profile/SettingSectionViewModel.swift [hunk 4]
```diff
@@ -0,0 +1,13 @@
+//
+//  SettingSectionViewModel.swift
+//  ecommerce-ios
+//
+//  Created by Nguyen M. Tam on 20/05/2021.
+//
+
+struct SettingSectionViewModel: Identifiable {
```
Comment (vnntsu) [inline ecommerce-ios/Sources/Profile/SettingSectionViewModel.swift]:
```suggestion
struct SettingsSectionViewModel: Identifiable {
```

File: ecommerce-ios/Sources/Profile/SettingSectionViewModel.swift [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+
+    let id: Int
+    let title: String
+    let cellViewModels: [SettingCellViewModel]
+}
```
Comment (vnntsu) [inline ecommerce-ios/Sources/Profile/SettingSectionViewModel.swift]: I suggest you add this extension

```swift
extension SettingsSectionViewModel {

    init(sectionType: SettingSectionType) {
        let cellViewModels = sectionType.settingTypes.map(SettingCellViewModel.init)
        self.init(
            id: sectionType.rawValue,
            title: sectionType.title,
            cellViewModels: cellViewModels
        )
    }
}
```

File: ecommerce-ios/Views/SettingCell/SettingCellViewModel.swift [hunk 6]
```diff
@@ -0,0 +1,13 @@
...
+//
+//  Created by Nguyen M. Tam on 20/05/2021.
+//
+
+struct SettingCellViewModel: Identifiable {
+
+    let id: Int
+    let imageString: String
+    let settingName: String
+}
```
Comment (vnntsu) [inline ecommerce-ios/Views/SettingCell/SettingCellViewModel.swift]: I suggest you to add this:

```swift
extension SettingsCellViewModel {

    init(settingType: SettingType) {
        self.init(
            id: settingType.rawValue,
            imageString: settingType.imageString,
            settingName: settingType.title
        )
    }
}
```

File: ecommerce-ios/Sources/Profile/ProfileScreen.swift [hunk 7]
```diff
@@ -9,8 +9,54 @@ import SwiftUI
...
+            sectionViewModels.append(
+                .init(
+                    id: settingSectionType.rawValue,
+                    title: settingSectionType.title,
+                    cellViewModels: cellViewModels
+                )
+            )
+        }
+        return sectionViewModels
+    }()
```
Comment (vnntsu) [inline ecommerce-ios/Sources/Profile/ProfileScreen.swift]:
```suggestion
    private let sectionViewModels: [SettingSectionViewModel] = {
        SettingSectionType.allCases.map(SettingSectionViewModel.init)
    }()
```

File: ecommerce-ios/Views/SettingCell/SettingSectionHeader.swift [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+                    .padding(.vertical, 8.0)
+                Spacer()
+            }
+        }
+        .frame(minHeight: 56.0)
+    }
+}
+
+struct SettingSectionHeader_Previews: PreviewProvider {
+    static var previews: some View {
```
Comment (vnntsu) [inline ecommerce-ios/Views/SettingCell/SettingSectionHeader.swift]: Add break line before

File: ecommerce-ios/Views/SettingCell/SettingCell.swift [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+            Image("common-icon/icon-disclosure")
+                .scaledToFill()
+                .frame(width: 8.0, height: 13.0)
+        }
+        .padding(.horizontal, 16.0)
+    }
+}
+
+struct SettingCell_Previews: PreviewProvider {
+    static var previews: some View {
```
Comment (vnntsu) [inline ecommerce-ios/Views/SettingCell/SettingCell.swift]: a break line before

**Review by blyscuit — APPROVED** [review]

**Review by vnntsu — APPROVED** [review]

**Pull request merged**